packages/context/src/__tests__/unit/handler-chain.unit.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  compareBindingsByTag,
+  Context,
+  filterByTag,
+  InvocationHandlerChain,
+  InvocationResult,
+  Next,
+} from '../..';
+
+describe('InvocationHandlerChain', () => {
+  let ctx: Context;
+  let handlerChain: InvocationHandlerChain;
+  let events: string[];
+
+  beforeEach(givenContext);
+
+  it('invokes handler functions', async () => {
+    givenInvocationHandlerChain();
+    const result = await handlerChain.invokeHandlers();
+    expect(events).to.eql([
+      'before-handler1',
+      'before-handler2',
+      'after-handler2',
+      'after-handler1',
+    ]);
+    expect(result).to.be.undefined();
+  });
+
+  it('honors return value', async () => {
+    givenInvocationHandlerChain({handler1: {}, handler2: {returnValue: 'ABC'}});
+    const result = await handlerChain.invokeHandlers();
+    expect(result).to.eql('ABC');
+  });
+
+  it('skips downstream handlers if next is not invoked', async () => {
+    givenInvocationHandlerChain({handler1: {skipNext: true}, handler2: {}});
+    await handlerChain.invokeHandlers();
+    expect(events).to.eql(['before-handler1', 'after-handler1']);
+  });
```

Comment:
I find these tests are very cryptic, it's difficult to understand what is being tested (e.g. what exactly does `skipNext` mean?) and why are we expecting the outcome encoded in assertions/expectations.

Can you find a better way how to structure these tests please? Even at the expense of a small amount of duplication.